### DISCUSSION TITLE
Clarification on providing vectors (glove, elmo, fasttext, etc) when training NER model

### DISCUSSION THREAD

**Discussion by CatarinaPC** [explosion/spaCy#5237] [Help: Coding & Implementations]
I have used spacy `pretrain` command to obtain pretrained weights and use them when training NER model.

**1 -** I have read somewhere, but I'm not sure that when training the model you can also provide other already trained vectors (I think I read something about elmo, glove and/or fasttext). Where can I find more information about this? Or is it possible for someone to give more information regarding this?

**2 -** Also, what is the difference between the parameters `--vectors` and `--init-tok2vec` ? What vectors does the first parameter refer to?

**Comment by svlandeg** [comment]
Hi and sorry for the late follow-up!

The `pretrain` command was still experimental in spaCy v2, and not very well documented. The good news is, that the recent [pre-release of spaCy v3](https://github.com/explosion/spaCy/releases/tag/v3.0.0rc1) comes with a revamped API for pretraining and swapping out `Tok2Vec` layers. The relevant documentation is here: https://nightly.spacy.io/usage/embeddings-transformers

The [pretrain](https://nightly.spacy.io/usage/embeddings-transformers) command is now run with the same [config file](https://nightly.spacy.io/usage/training#config) you'd also use during training, making the whole process much more robust and transparent. You can get started with these config files using [`init config`](https://nightly.spacy.io/api/cli#init-config) and setting the pretraining flag, e.g.:
```
python -m spacy init config config.cfg --lang en --pipeline ner --optimize accuracy --pretraining
```
which will produce a config file with (hopefully) sensible defaults.

Hope that's useful to you! I will preliminary close this issue, but feel free to comment here or open a new one if you run into any issues.
